c_tests/tests/compile_many.c

```diff
@@ -0,0 +1,29 @@
+// Compiler:
+// Run-time:
+
+// Check that compiling and running multiple traces in sequence works.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk_testing.h>
+
+void trace(void) {
+  void *tt = __yktrace_start_tracing(HW_TRACING);
+  int res = 1 + 1;
```

Comment:
At the risk of flogging a dead horse, this line is likely to be removed entirely by the optimiser.

Reply:
Once we can pass arguments into the trace we will go around and update all tests. Before then we don't have much choice here.

Reply:
OK, but let's at least add FIXME to each one that needs fixing.

Reply:
Will do.